src/av/av_detect.erl

```diff
@@ -2,43 +2,75 @@
 -export([is_infected/2]).
 -include("av_recs.hrl").
 
-%%% Perform the malware detection.
+%%% Perform the illicit content detection.
 %%% Given it's task, we should optimise this as much as possible.
 
-%% Given a filename and a set of signatures, test whether the file
-%% is infected. If not, return false, if it is, return true and
-%% the matching sigs.
-%% Only files can be infected, return false for directories.
-% is_infected(FileName, Sigs) ->
-% 	case try_read(FileName) of
-% 		error -> pass;
-% 		Bin -> do_is_infected(FileName, Bin, Sigs)
-% 	end.
+%% Given a list of binaries and a set of signatures, test whether some binaries
+%% contain illicit content or have illicit hash. If not, return false, if yes, return true and
+%% the matching signatures.
+is_infected(Binaries, {Sigs, BinaryPattern}) ->
+	HashSigs = hash_sigs(Sigs),
+	MatchedHashSigs = case HashSigs of
```

Comment:
Can't we replace this whole case with:
`MatchedHashSigs = full_check(Binaries, HashSigs),`?

Reply:
✅